gui/src/app/state/spend/step.rs

```diff
@@ -166,18 +171,22 @@ impl Recipient {
             && self.amount.valid
     }
 
-    fn update(&mut self, message: view::CreateSpendMessage) {
+    fn update(&mut self, network: Network, message: view::CreateSpendMessage) {
         match message {
             view::CreateSpendMessage::RecipientEdited(_, "address", address) => {
                 self.address.value = address;
-                if self.address.value.is_empty() {
-                    // Make the error disappear if we deleted the invalid address
-                    self.address.valid = true;
-                } else if Address::from_str(&self.address.value).is_ok() {
-                    self.address.valid = true;
+                if let Ok(address) = Address::from_str(&self.address.value) {
+                    self.address.valid = if network == Network::Bitcoin {
+                        address.network == Network::Bitcoin
+                    } else {
+                        address.network != Network::Bitcoin
+                    };
```

Comment:
It's sufficient for safety (it's fine if they mixup test networks) but not for correctness: regtest addresses have their own prefix.

Reply:
I think I made the error somewhere else also in the gui, which is not bad, just wrong.